File: bin/src/poc/vessel_py/vessel.py
#!/usr/bin/env python3
"""
Vessel - 動的スクリプト実行コンテナ
標準入力から受け取ったPythonコードを実行する器
"""
import sys

def main():
    # 標準入力からスクリプトを読み込む
    script = sys.stdin.read()
    
    # 実行環境を準備
    context = {
        '__name__': '__main__',
        'vessel': True  # vessel内で実行されていることを示すフラグ
    }
    
    # スクリプトを実行
    try:
        exec(script, context)
    except Exception as e:
        print(f"Error executing script: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: bin/src/poc/vessel_ts/vessel.ts
#!/usr/bin/env bun
/**
 * Vessel for Bun - Dynamic script execution container
 * For TypeScript support, use .ts files or transpile first
 */

const script = await Bun.stdin.text();

// Simple approach - eval for JavaScript only
// For TypeScript, use Bun's file-based execution
try {
  // For JavaScript, use eval with async wrapper for top-level await
  const AsyncFunction = Object.getPrototypeOf(async function(){}).constructor;
  const func = new AsyncFunction(script);
  await func();
} catch (error) {
  console.error("Error executing script:", error);
  process.exit(1);
}